fusor-ember-cli/app/controllers/assign-nodes.js

```diff
@@ -30,12 +30,14 @@ export default Ember.Controller.extend(DeploymentControllerMixin, NeedsDeploymen
     var params = this.get('model.plan.parameters');
     var self = this;
     var value = null;
-    this.get('model.plan.roles').forEach(function(role) {
-      value = self.getParamValue(role.get('flavorParameterName'), params);
-      if (value === 'baremetal' || Ember.isNone(value)) {
-        unassignedRoles.addObject(role);
-      }
-    });
+    if (Ember.isPresent(this.get('model.plan.roles'))) {
```

Comment:
This PR also modifies the openstack tabs disabled/enabled state, and I was getting errors such defined @forEach of null, so I wrapped this in an `ifPresent` statement to ensure there is an empty [] and not null